test/dark-gravity-wave-test.js

```diff
@@ -1,126 +1,286 @@
+            const result = dgw.isValidTarget(highTargetInt, blocks.slice(0, 10));
+            expect(result).to.equal(true);// TODO change on false?
+        });
+
+        it('should be invalid maxTarget = 0x207fffff as int', () => {
+            const highTargetInt = 545259519;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(false);
+        });
+
+        it('should be valid for maxTarget > 0x207fffff', () => {
+            const highTargetInt = 0x20800000;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(true);
+        });
+
+        it('should be valid for maxTarget > 0x207fffff as int', () => {
+            const highTargetInt = 0x20800000;
+            const result = dgw.isValidTarget(highTargetInt, blocks);
+            expect(result).to.equal(true);
+        });
+
+        it('should be valid when allHeaders is empty array', () => {
+            const highTargetInt = 0x20800000;
+            const result = dgw.isValidTarget(highTargetInt, [{}]);
+            expect(result).to.equal(true);
+        });
+
+        it('should throw error for allHeaders is {}', () => {
+            const highTargetBits = 0x208fffff;
+            expect(() => {
+                dgw.isValidTarget(highTargetBits, {});
+            }).to.throw('allHeaders.slice is not a function');
+        });
+
+        it('should throw error for allHeaders is string', () => {
+            // TODO not very good that we allow string although it's length < 25 and ignored in getTarget
+            const highTargetBits = 0x1b0777d4;
+            const result = dgw.isValidTarget(highTargetBits, 'my_string');
+            expect(result).to.equal(true);
+        });
+
+        it('should throw error for allHeaders is string > 25 chars', () => {
+            const highTargetBits = 0x1b0777d4;
+            expect(() => {
+                dgw.isValidTarget(highTargetBits, 's'.repeat(26));
+            }).to.throw('allHeaders.slice(...).reverse is not a function');
+        });
+
+        it('should throw error for signed int', () => {
+            expect(() => {
+                dgw.isValidTarget(-1, blocks);
+            }).to.throw('byte array longer than desired length');
+        });
+
+        it('should throw error when compact is undefined', () => {
+            expect(() => {
+                dgw.isValidTarget(undefined, blocks);
+            }).to.throw('Compact of type undefined not supported');
+        });
+
+        it('should throw error when compact is boolean', () => {
+            expect(() => {
+                dgw.isValidTarget(true, blocks);
+            }).to.throw('Compact of type boolean not supported');
+        });
+
+        it('height values ignored in calculation in getTarget function', () => {
+            //TODO is it true?
+            //and we can pass any blocks height: invalid, duplicates, not ordered...?
```

Comment:
yes I think that is fine.  Consensus rules do not consider height and its up to the dev to provide correct `allHeaders`.  Since DGW is stateless it cannot determine if the previous 25 headers provided is conforming to consensus based on earlier blocks.